packages/saber-plugin-generate-feed/lib/index.js

```diff
@@ -1,58 +1,102 @@
+    options
+  )
+
   const { siteConfig } = api.config
   if (!siteConfig.url) {
     throw new Error(`siteConfig.url is required for saber-plugin-generate-feed`)
   }
-  const feedUrl = url.resolve(siteConfig.url, 'feed.json')
+
+  // All type of feed links
+  const feedLinks = options.feeds
+    ? options.feeds
+    : {
+        json: urljoin(siteConfig.url, 'feed.json'),
+        atom: urljoin(siteConfig.url, 'atom.xml')
+      }
 
   api.hooks.afterGenerate.tapPromise(ID, async () => {
+    // Prepare posts
     const posts = []
-
     for (const page of api.pages.values()) {
       if (page.attributes.type === 'post' && !page.attributes.draft) {
         posts.push({
           title: page.attributes.title,
           id: page.attributes.permalink,
-          url: url.resolve(siteConfig.url, page.attributes.permalink),
-          content_html: page.content,
-          date_published: new Date(page.attributes.createdAt),
-          date_modified: page.attributes.updatedAt
+          link: urljoin(siteConfig.url, page.attributes.permalink),
+          description: page.attributes.description,
+          content: page.content,
+          date: page.attributes.updatedAt,
+          published: page.attributes.createdAt
         })
       }
     }
+    // Order by published
     const items = posts
       .sort((a, b) => {
-        return a.date_published > b.date_published ? -1 : 1
+        return b.published - a.published
       })
-      .slice(0, limit)
-    if (items.length > 0) {
-      const feed = {
-        version: 'https://jsonfeed.org/version/1',
-        title: siteConfig.title,
-        description: siteConfig.description,
-        author:
-          typeof siteConfig.author === 'string'
-            ? {
-                name: siteConfig.author
-              }
-            : siteConfig.author,
-        home_page_url: siteConfig.url,
-        feed_url: feedUrl,
-        items
-      }
-      const { log } = api
-      const { fs } = api.utils
-      log.info(`Generating feed.json`)
+      .slice(0, options.limit)
+
+    // Feed instance
+    const feed = new Feed({
+      title: siteConfig.title,
+      description: siteConfig.description,
+      id: siteConfig.url,
+      link: siteConfig.url,
+      copyright: options.copyright,
+      generator: options.generator,
+      author:
+        typeof siteConfig.author === 'string'
+          ? {
+              name: siteConfig.author
+            }
+          : siteConfig.author,
+      feedLinks
+    })
+
+    // Add posts to feed
+    items.forEach(post => {
+      feed.addItem(post)
+    })
+
+    const { log } = api
+    const { fs } = api.utils
+
+    // JSON
+    if (feedLinks.json) {
+      log.info('Generating json feed')
+      await fs.outputFile(
+        api.resolveCwd(
+          `.saber/public/${feedLinks.json.replace(siteConfig.url, '')}`
+        ),
+        feed.json1(),
+        'utf8'
+      )
+    }
+
+    // Atom and RSS2 (Note: rss2 also uses atom to generate)
```

Comment:
> rss2 also uses atom to generate

why?

Reply:
The feed library will use atom field to generate rss2 feed if the atom feed config exists, see https://github.com/jpmonette/feed/blob/master/src/rss2.ts#L67 , IDKW.